openml/tasks/task.py

```diff
@@ -36,14 +36,24 @@ class OpenMLTask(OpenMLBase):
 
     Parameters
     ----------
+    task_id : Optional[int]
```

Comment:
I think the typing below is incorrect. This should be `Union[int, None]` (if possible) as it is not an optional parameter for this constructor. Hence, I would change it here and below to reflect this.